src/resources.cc

```diff
@@ -0,0 +1,248 @@
+
+            auto req = res_data->video_requirements;
+
+            if (req->width > 0)
+              {
+                capture_params.push_back(cv::CAP_PROP_FRAME_WIDTH);
+                capture_params.push_back(req->width);
+                this->_logger->info("Requested width={}", req->width);
+              }
+            if (req->height > 0)
+              {
+                capture_params.push_back(cv::CAP_PROP_FRAME_HEIGHT);
+                capture_params.push_back(req->height);
+                this->_logger->info("Requested height={}", req->height);
+              }
+            if (req->fps > 0)
+              {
+                capture_params.push_back(cv::CAP_PROP_FPS);
+                capture_params.push_back(req->fps);
+                this->_logger->info("Requested fps={}", req->fps);
+              }
+            std::string req_fourcc = req->fourcc->std_str();
+            if (req_fourcc.size() == 4)
+              {
+                capture_params.push_back(cv::CAP_PROP_FOURCC);
+                capture_params.push_back(
+                    cv::VideoWriter::fourcc(req_fourcc[0], req_fourcc[1],
+                                            req_fourcc[2], req_fourcc[3]));
+                this->_logger->info("Requested fourcc={}", req_fourcc);
+              }
+            else if (req_fourcc.size() != 0)
+              {
+                this->_logger->warn("Requested invalid fourcc: {}",
+                                    req_fourcc);
+              }
+          }
+      }
+    catch (...)
+      {
+        // pass
+      }
+
+    bool in_is_gst_pipeline = boost::algorithm::ends_with(uri, "! appsink");
+
+    // set params
+    for (size_t i = 0; i < capture_params.size(); i += 2)
+      {
+        _capture.set(capture_params.at(i), capture_params.at(i + 1));
+      }
+
+    // open capture
+    if (cam_id >= 0)
+      {
+        this->_logger->info("Opening VideoCapture on camera {}", cam_id);
+        _capture.open(cam_id, reader_backend);
+      }
+    else if (in_is_gst_pipeline)
+      {
+        this->_logger->info("Opening VideoCapture with pipeline {}", uri);
+        _capture.open(uri, cv::CAP_GSTREAMER);
+      }
+    else
+      {
+        this->_logger->info("Opening VideoCapture on video {}", uri);
+        _capture.open(uri, reader_backend);
+      }
+
+    if (!_capture.isOpened())
+      {
+        this->_logger->error("VideoCapture could not be opened");
+        throw ResourceBadParamException("Video with uri \"" + uri
+                                        + "\" could not be opened");
+      }
+
+    int fps = _capture.get(cv::CAP_PROP_FPS);
+    cv::Size dims((int)_capture.get(cv::CAP_PROP_FRAME_WIDTH),
+                  (int)_capture.get(cv::CAP_PROP_FRAME_HEIGHT));
+    auto frame_count = uint32_t(_capture.get(cv::CAP_PROP_FRAME_COUNT));
+    int fourcc = static_cast<int>(_capture.get(cv::CAP_PROP_FOURCC));
+    this->_logger->info("Video properties: {}x{} - {} fps, {} frames, enc={}",
```

Comment:
Are we returning this info as JSON output to the API call ? If not, could be useful to do so!

Reply:
Done